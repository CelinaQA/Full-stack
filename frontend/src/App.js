import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';

import Navigation from './components/Header/Navigation';
import Home from './components/Header/Home';
import MyTasks from './components/Header/MyTasks';


function App() {
  return (
    <div className="App">
      <Router>
        <Navigation />
        <Switch>
          <Route path="/" exact>
            <Home />
          </Route>
          <Route path="/task">
            <MyTasks />
          </Route>
          {/* <Route>
            <NotFound />
          </Route> */}
        </Switch>
      </Router>
    </div>
  );
}

export default App;
