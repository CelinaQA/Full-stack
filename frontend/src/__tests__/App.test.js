import { render, screen } from '@testing-library/react';
import App from '../App';
import '@testing-library/jest-dom';
import { expect, it } from '@jest/globals';
import { create } from 'react-test-renderer';

test('renders learn react link', () => {
  render(<App />);
  const linkElement = screen.getByText(/Home/i);
  expect(linkElement).toBeInTheDocument();
});

describe(`snapshot test of app`, () => {

  it(`should match the snapshot every time it is rendered`, () => {
    const instance = create(<App />).toJSON();
    expect(instance).toMatchSnapshot();
  })
})
