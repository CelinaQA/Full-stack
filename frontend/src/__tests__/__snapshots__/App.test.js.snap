// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test of app should match the snapshot every time it is rendered 1`] = `
<div
  className="App"
>
  <nav
    className="navbar navbar-expand-md navbar-dark bg-dark"
  >
    <ul
      className="mr-auto navbar-nav"
    >
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li
        className="nav-item"
      >
        <a
          className="nav-link"
          href="/task"
          onClick={[Function]}
        >
          My Tasks
        </a>
      </li>
    </ul>
  </nav>
  <div
    className="container"
  >
    <h2>
      Welcome
    </h2>
  </div>
</div>
`;
